vllm/model_executor/models/minicpm.py

```diff
@@ -437,6 +438,73 @@ def forward(
         hidden_states = self.norm(hidden_states)
         return hidden_states
 
+    def load_weights(self, weights: Iterable[Tuple[str,
+                                                   torch.Tensor]]) -> Set[str]:
+        stacked_params_mapping = [
+            # (param_name, shard_name, shard_id)
+            ("qkv_proj", "q_proj", "q"),
+            ("qkv_proj", "k_proj", "k"),
+            ("qkv_proj", "v_proj", "v"),
+            ("gate_up_proj", "gate_proj", 0),
+            ("gate_up_proj", "up_proj", 1),
+        ]
+        expert_params_mapping = [
+            # (param_name, weight_name, expert_id)
+            ("ws" if weight_name in ["w1", "w3"] else "w2s",
+             f"experts.{expert_id}.{weight_name}.weight", expert_id)
+            for expert_id in range(self.num_experts)
+            for weight_name in ["w1", "w2", "w3"]
+        ]
```

Comment:
I'm thinking about refactoring `MiniCPMMoE` to inherit from `FusedMoE` like Aria, so that maybe we can reduce some duplicated codes. 

Do you know the reason why implement `MiniCPMMoE` manually instead of using `FusedMoE`?

Reply:
Not sure, I wasn't there when MiniCPM was originally added to vLLM. I can try refactoring that.

Reply:
On further inspection, it seems that the nesting structure is a bit different, In particular, the `router_logits` computation is inside `MiniCPMMoE`, whereas `FusedMoE` expects `router_logits` to be computed outside.

Reply:
OK, let's keep it as it looks now!